scripts/list-langs.py

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python3
+"""Prints the list of languages tracked in the repository."""
+
+from argparse import ArgumentParser
+import git
+import os
+
+def list_tracked_dirs(git_root):
+    """Get the list of tracked directories in the repository in *git_root*"""
+    repo = git.Repo(git_root)
+    tracked_files = repo.git.ls_tree("--full-tree", "--name-only", "HEAD").splitlines()
+    return tracked_files
+
+def filter_ignored_entries(file_list: list, ignored_entries: list) -> list:
+    """Filter out the ignored entries, to return the valid languages codes"""
+    filtered_files = [file_name for file_name in file_list if file_name not in ignored_entries]
+    return filtered_files
+
+if __name__ == "__main__":
+    parser = ArgumentParser(description=__doc__)
+    parser.add_argument('--git-root', default=os.getcwd(), required=False)
+    args = parser.parse_args()
+
+    git_root = git.Repo(args.git_root, search_parent_directories=True).working_dir
+    ignored_entries = ['.tx', 'pot']
+
+    tracked_files = list_tracked_dirs(git_root)
+    languages = filter_ignored_entries(tracked_files, ignored_entries)
```

Comment:
nit: assuming we don't have any builds artifacts or similar in the repositories checkout, and clean git tree, following would be a "lightweight" alternative
```suggestion
"""Prints the list of languages in a repository."""

from argparse import ArgumentParser
from pathlib import Path

import os

def list_dirs(root):
    """Get the list of not empty and not hidden directories in the repository in *root*"""
    return [el.name for el in root.iterdir() if el.is_dir() and el.iterdir() and not el.name.startswith(".")]

def filter_ignored_entries(file_list: list, ignored_entries: list) -> list:
    """Filter out the ignored entries, to return the valid languages codes"""
    filtered_files = [file_name for file_name in file_list if file_name not in ignored_entries]
    return filtered_files

if __name__ == "__main__":
    parser = ArgumentParser(description=__doc__)
    parser.add_argument('--root', default=os.getcwd(), required=False)
    args = parser.parse_args()

    root = Path(args.root)
    ignored_entries = ['pot']

    dirs = list_dirs(root)
    languages = filter_ignored_entries(dirs, ignored_entries)
```

Reply:
I would like to assure the directory has .po files inside, do you have an idea how we can achieve that?

Reply:
Adding ` and any(el.glob("**/*.po")` to the condition should achieve that.

By the way a self-correction: `any(el.iterdir())` should used be used instead of `el.iterdir()` to assure that directory is not empty, but it will be superfluous after checking for `.po` files specifically.

Reply:
Moreover `el.is_dir()` is also superfluous with the directory content checks, therefore all conditions can be now shortened.

```python
return [el.name for el in root.iterdir() if any(el.glob("**/*.po")) and not el.name.startswith(".")]
```

Reply:
Thanks a lot! Applied this in #37